sources/lib/util/steps.mli

```diff
@@ -0,0 +1,21 @@
+type incr_kind =
```

Comment:
There is a missing header for this file.
Also, all interface files should start with doc comments of the form:
```ocaml
(* ... header ..... *)

(** Module_Name

    Long description of the rationale and intent for the module, generic explanations about the way it behaves, about
    some of the design choices about it, etc...
*)
```

Also, some ocamldoc titles would be useful, e.g.:
```ocaml
(** {1 Steps counters} *)
```